sips/sips/sip-009-nft-standard.md

```diff
@@ -0,0 +1,101 @@
+
+# License and Copyright
+
+This SIP is made available under the terms of the Creative Commons CC0 1.0 Universal license, available at https://creativecommons.org/publicdomain/zero/1.0/
+This SIP’s copyright is held by the Stacks Open Internet Foundation.
+
+# Introduction
+
+Tokens are digital assets that are registered on the Stacks blockchain through a smart contract. A non-fungible token (NFT) is a token that is globally unique and that users can identify through its identifier. The smart contract that registers the NFTs defines a name for the group of NFTs.
+
+NFTs are enumerated, the id starts at 1 and the current last id is provided by the smart contract.
+
+# Specification
+
+NFT smart contract shall implement the trait defined at `ST314JC8J24YWNVAEJJHQXS5Q4S9DX1FW5Z9DK9NT.nft-trait` as well as satisfy the additional conditions.
+The trait has three functions:
+* `get-last-token-id` does not take any arguments and returns the highest number that is used as an identifier for any NFT. This is the upper limit when iterating through all NFTs.
+* `get-token-uri` takes an NFT identifier and returns a response containg a URI pointing to meta data of the given identifier. The URI is wrapped as an optional, which means if the corresponding NFT exists or the contract does not maintain meta data the response is `(ok none)`, otherwise, e.g. `(ok (some "https://example.com"))`. The length of the returned uri is limited to 256. The specification of the meta data should be covered in a separate SIP.
+* `get-owner` takes an NFT identifier and returns a response containing the principal owning the NFT for the given identifier. The principal is wrapped as an optional, which means if the corresponding NFT does not exists the response is `(ok none)`, otherwise, e.g. `(ok (some 'ST12...))`. The owner can be a contract principal.
+* `transfer` takes an NFT identifier, a sender principal and a receiver principal. The function changes the ownership of the NFT for the given identifier. The change has to be reflected in the `get-owner` function, for details see implementation rules.
+
+## Trait
+
+```
+(define-trait stacks-token-nft-standard-v1
+  (
+    ;; Token ID, limited to uint range
+    (get-last-token-id () (response uint uint))
+
+    ;; URI for meta data about the token 
+    (get-token-uri (uint) (response (optional (string-ascii 256)) uint))
+
+     ;; Owner of given token identifier
+    (get-owner (uint) (response (optional principal) uint))
+
+    ;; Transfer from to
+    (transfer (uint principal principal) (response bool (tuple (kind (string-ascii 32)) (code uint))))
+  )
+)
+```
+
+## Implementation rules
+
+1. Contracts must use a least one NFT asset. A post condition with deny mode and without any NFT condition about a changed owner must fail for `transfer` function calls.
```

Comment:
```suggestion
1. Contracts must use a least one NFT asset. A postcondition with deny mode and without any NFT condition about a changed owner must fail for `transfer` function calls.
```

Reply:
using `post-conditions` now as in docs.blockstack.org